### ISSUE TITLE
`VersionMismatch` error raised when attempting to publish new content

### ISSUE THREAD

**Issue by reidcooper** [contentful/contentful-management.js#191]
I am using the contentful migration library to modify a bunch of content then publish it immediately after. Updating the content works properly but when I attempt to publish, I am given a `VersionMismatch`. What is the best practice?

```
    const contentful = require('contentful-management');

    const environment = process.env.CONTENTFUL_MIGRATION_ENV || 'review-for-qa';

    const client = contentful.createClient({
      accessToken: accessToken,
      environment: environment
    });

    client.getSpace(spaceId)
          .then((space) => space.getEnvironment(environment))
          .then((env) => env.getEntry(parentId))
          .then((entry) => {
            entry.fields.associatedIcon['en-US'] = {
              "sys": {
                "id": new_e5_reference,
                "type": "Link",
                "linkType": "Entry"
              }
            };

            entry.update();
          })
          .then((entry) => console.log(`Entry ${entry.sys.id} updated.`))
          .catch(console.error);

    client.getSpace(spaceId)
          .then((space) => space.getEnvironment(environment))
          .then((env) => env.getEntry(parentId))
          .then((entry) => {
            return entry.publish();
          })
          .then((entry) => console.log(`Entry ${entry.sys.id} published.`))
          .catch(console.error);
```

**Comment by Khaledgarbaya** [comment]
Hey @reidcooper,

Maybe something like this.

```diff
const contentful = require('contentful-management');

    const environment = process.env.CONTENTFUL_MIGRATION_ENV || 'review-for-qa';

    const client = contentful.createClient({
      accessToken: accessToken
-     environment: environment
    });

    client.getSpace(spaceId)
          .then((space) => space.getEnvironment(environment))
          .then((env) => env.getEntry(parentId))
          .then((entry) => {
            entry.fields.associatedIcon['en-US'] = {
              "sys": {
                "id": new_e5_reference,
                "type": "Link",
                "linkType": "Entry"
              }
            };

-            entry.update();
+            return entry.update()
          })
+          .then((entry) => console.log(`Entry ${entry.sys.id} updated.`) || entry.publish())
          .then((entry) => console.log(`Entry ${entry.sys.id} published.`))
          .catch(console.error);
- client.getSpace(spaceId)
-          .then((space) => space.getEnvironment(environment))
-          .then((env) => env.getEntry(parentId))
-          .then((entry) => {
-            return entry.publish();
-          })
-          .then((entry) => console.log(`Entry ${entry.sys.id} published.`))
-          .catch(console.error);
```
The only thing I see is you didn't return the entry.update() promise.
Can you try to use the code I provided an let me know if you still have an issue

**Comment by reidcooper** [comment]
I tried that out. That indeed worked. My question(s):
1. why do i have to return `entry.update()`?
2. `console.log(`Entry ${entry.sys.id} updated.`) || entry.publish())` why does that have to be in the same line? Or it doesn't matter if its broken down into two `then` statements?

**Comment by Khaledgarbaya** [comment]
Hi @reidcooper,

> why do i have to return entry.update()?

`entry.update()is a promise so you need to return it to [chain](https://javascript.info/promise-chaining) the promises.

> console.log(Entry ${entry.sys.id} updated.) || entry.publish()) why does that have to be in the same line? Or it doesn't matter if it's broken down into two then statements?

the element you get in then is dependent always on the previously returned value.
in your previous code, you had 

```js
.then((entry) => console.log(`Entry ${entry.sys.id} updated.`)) 
```
console.log() function returns undefined after it runs so the next then will receive `undefined`. the reason I did `console.log(...) || entry.publish()` is a trick to log something and at the same time return the `entry.publish()`

TlDr;

The `||` operator returns the left operand if it evaluates as true, otherwise it evaluates and returns the right operand. In other words, `a || b` is equivalent to `a ? a : b` except that a is only evaluated once


since `console.log` is `undefined`, aka falsy value,  and `entry.publish()` is a `promise`, truthy value, js will pick the promise.

one other alternative can be 

```js
then(entry => {
console.log(`Entry ${entry.sys.id} updated.`)
return entry.publish()
})
```

I hope that helped.

Best,
Khaled

**Comment by reidcooper** [comment]
That makes sense, thank you very much!
